tests/compute/half-calc.slang

```diff
@@ -0,0 +1,32 @@
+//TEST(compute):COMPARE_COMPUTE:-dx12 -compute -use-dxil -profile cs_6_2 -render-features half
+//TEST(compute):COMPARE_COMPUTE:-vk -compute -use-dxil -profile cs_6_2 -render-features half
+//TEST_INPUT:ubuffer(data=[0 0 0 0], stride=4):dxbinding(0),glbinding(0),out
+
+// Test for doing a calculation using half
+
+RWStructuredBuffer<float> outputBuffer;
+
+[numthreads(4, 1, 1)]
+void computeMain(uint3 dispatchThreadID : SV_DispatchThreadID)
+{
+    uint tid = dispatchThreadID.x;
+    
+    //half2 v0 = { 1, -2 };
+    //half2 v1 = { -2, 4 };
+    
+    //half2 v2 = (v0 * 2.0f + v1);
+    
+    // This should work (it compiles on dxc, but slang doesn't seem to have overloads yet)
+    //half offset = length(v2);
```

Comment:
I just went to double check, and it seems like `half` supports the `__BuiltinFloatingPointType` interface, and `length` should be defined for any vector over a floating-point type, so this definitely seems weird.